deploy/docker/docker-entrypoint.sh

```diff
@@ -1,24 +1,26 @@
-#!/bin/sh
+#!/bin/bash
 ## EMQ docker image start script
 # Huang Rui <vowstar@gmail.com>
 # EMQ X Team <support@emqx.io>
 
 ## Shell setting
-if [[ ! -z "$DEBUG" ]]; then
+if [[ -n "$DEBUG" ]]; then
     set -ex
 else
     set -e
 fi
 
+shopt -s nullglob
+
 ## Local IP address setting
 
-LOCAL_IP=$(hostname -i |grep -E -oh '((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])'|head -n 1)
+LOCAL_IP=$(hostname -i | grep -oE '((25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])\.){3}(25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])' | head -n 1)
```

Comment:
Format and improve the regex.
The `-h` for grep is unusual because it read input from stdin instead of file.